ocamldoc/odoc_sig.ml

```diff
@@ -282,23 +282,26 @@ module Analyser =
           | Ptyp_object (fields, _) ->
             let rec f = function
               | [] -> []
-              | ({txt=""},_,_) :: _ ->
+              | Otag ({txt=""},_,_) :: _ ->
                 (* Fields with no name have been eliminated previously. *)
                 assert false
-
-              | ({txt=name}, _atts, ct) :: [] ->
+              | Otag ({txt=name}, _atts, ct) :: [] ->
                 let pos = Loc.ptyp_end ct in
                 let (_,comment_opt) = just_after_special pos pos_end in
                 [name, comment_opt]
-              | ({txt=name}, _atts, ct) :: ((_name2, _atts2, ct2) as ele2) :: q ->
+              | Otag ({txt=name}, _, ct) ::
+                  ((Oinherit ct2) as ele2) :: q
+              | Otag ({txt=name}, _, ct) ::
+                  ((Otag (_, _, ct2)) as ele2) :: q ->
```

Comment:
It might be worthwhile to fusion the two cases here into 
```OCaml
| (Otag ({txt=name}, _, ct) ) :: ((Oinherit ct2 | Otag (_, _, ct2) ) as ele2) :: q
```